pkg/email/email_test.go

```diff
@@ -0,0 +1,216 @@
+package email_test
+
+import (
+	_ "embed"
+	"errors"
+	"math"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/email"
+	"github.com/treeverse/lakefs/pkg/testutil"
+	"golang.org/x/time/rate"
+	"gopkg.in/gomail.v2"
+)
+
+var (
+	//go:embed invite_user.golden
+	inviteUserGolden string
+
+	//go:embed reset_password.golden
+	resetPasswordGolden string
+)
+
+type Dialer struct {
+	Message []*gomail.Message
+}
+
+func (d *Dialer) DialAndSend(m ...*gomail.Message) error {
+	d.Message = m
+	return nil
+}
+
+func TestNewEmailer(t *testing.T) {
+	tests := []struct {
+		name     string
+		smtpHost string
+		sender   string
+		expected error
+	}{
+		{name: "missing_smtp_host", smtpHost: "", sender: "bar", expected: nil},
+		{name: "missing_sender", smtpHost: "foo", sender: "", expected: email.ErrSenderMisconfigured},
+		{name: "missing_smtp_host_and_sender", smtpHost: "", sender: "", expected: nil},
+		{name: "misconfigured_sender", smtpHost: "test.com", sender: "bar", expected: email.ErrSenderMisconfigured},
+		{name: "noraml", smtpHost: "test.com", sender: "alice@testing.com", expected: nil},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := email.Params{
+				SMTPHost: tt.smtpHost,
+				Sender:   tt.sender,
+			}
+			_, err := email.NewEmailer(p)
```

Comment:
Add a check that if err is nil - we get an emailer

Reply:
Done